seleniumJSEnv/src/main/scala/org/scalajs/jsenv/selenium/CustomFileMaterializer.scala

```diff
@@ -0,0 +1,57 @@
+package org.scalajs.jsenv.selenium
+
+import java.io.File
+import java.net.URL
+
+import org.scalajs.core.tools.io.{IO, VirtualTextFile, WritableFileVirtualTextFile}
+
+/** Materializes files on the filesystem and specifies a custom url to access stored files.
+  * This can be used to bypass cross origin access policies as shown below.
+  *
+  * @param fsRoot Location on the filesystem where to store the generated files
+  * @param webRoot Corresponding url to access the files
+  *
+  * @example
+  *
+  * The following illustrates how to configure a project such that the browser fetches
+  * files by http:// instead of file://.
+  * This example assumes a local webserver is running and serving the ".tmp"
+  * directory at http://localhost:8080
+  *
+  * <pre>
+  *   jsSettings(
+  *   // ...
+  *     jsEnv := new SeleniumJSEnv(org.scalajs.jsenv.selenium.Firefox)
+  *         .withMaterializer(new SpecificFileMaterializer(".tmp", "http://localhost:8080"))
+  * )
+  * </pre>
+  */
```

Comment:
Please follow our coding style for all comments (issue with allignment of `*`): 
https://github.com/scala-js/scala-js/blob/master/CODINGSTYLE.md#scaladoc